lib/experimental/new_embed.dart

```diff
@@ -402,27 +402,35 @@ class NewEmbed {
     window.parent.postMessage({'sender': 'frame', 'type': 'ready'}, '*');
   }
 
-  String get gistId {
+  String _getQueryParam(String key) {
     final url = Uri.parse(window.location.toString());
 
-    if (url.hasQuery &&
-        url.queryParameters['id'] != null &&
-        isLegalGistId(url.queryParameters['id'])) {
-      return url.queryParameters['id'];
+    if (url.hasQuery && url.queryParameters[key] != null) {
+      return url.queryParameters[key];
     }
 
     return '';
   }
 
-  String get sampleId {
-    final url = Uri.parse(window.location.toString());
+  // ID for a GitHub gist that should be loaded into the editors.
+  String get gistId {
+    final id = _getQueryParam('id');
+    return isLegalGistId(id) ? id : '';
```

Comment:
Out of curiosity: why an empty string instead of `null`?

Reply:
Mostly it's a "that's how it was done in the codebase when I arrived" thing.